src/SqlPersistence/Config/SqlPersistenceConfig_Connection.cs

```diff
@@ -19,14 +19,24 @@ public static void ConnectionBuilder(this PersistenceExtensions<SqlPersistence>
                 .Set("SqlPersistence.ConnectionBuilder", connectionBuilder);
         }
 
-        internal static Func<DbConnection> GetConnectionBuilder(this ReadOnlySettings settings)
+        internal static Func<DbConnection> GetConnectionBuilder(this ReadOnlySettings settings, Type storageType)
         {
-            if (settings.TryGet("SqlPersistence.ConnectionBuilder", out Func<DbConnection> value))
+            if (settings.TryGet($"SqlPersistence.ConnectionBuilder.{storageType.Name}", out Func<DbConnection> value))
+            {
+                return value;
+            }
+            if (settings.TryGet("SqlPersistence.ConnectionBuilder", out value))
             {
                 return value;
             }
             throw new Exception("ConnectionBuilder must be defined.");
```

Comment:
I know this wasn't changed but maybe we could provide a better exception here, and also include the storage type it's trying to find.

Reply:
> also include the storage type it's trying to find.

👍